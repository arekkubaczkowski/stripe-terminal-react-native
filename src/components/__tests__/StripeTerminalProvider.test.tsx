import * as React from 'react';
import { StripeTerminalProvider } from '../StripeTerminalProvider';
import { fireEvent, render } from '@testing-library/react-native';
import { Text, TouchableOpacity } from 'react-native';
import { useStripeTerminal } from '../../hooks/useStripeTerminal';

jest.mock(
  '../../../node_modules/react-native/Libraries/EventEmitter/NativeEventEmitter'
);

describe('StripeTerminalProvider.tsx', () => {
  it('renders children correctly', () => {
    const tokenProvider = jest.fn();
    const { toJSON, findByText } = render(
      <StripeTerminalProvider tokenProvider={tokenProvider} logLevel="error">
        <Text>test text</Text>
      </StripeTerminalProvider>
    );
    const childText = findByText('test text');
    expect(toJSON()).toMatchSnapshot();
    expect(childText).toBeTruthy();
  });

  it('ensure if tokenProvider is not called on render', () => {
    const tokenProvider = jest.fn();
    render(
      <StripeTerminalProvider
        tokenProvider={tokenProvider}
        logLevel="error"
        children={<></>}
      />
    );
    expect(tokenProvider).not.toBeCalled();
  });

  it('trigger tokenProvider on init', () => {
    const tokenProvider = jest.fn().mockReturnValue('_token');

    const ChildImpl = () => {
      const { initialize } = useStripeTerminal();

      return (
        <TouchableOpacity onPress={() => initialize()}>
          <Text>init</Text>
        </TouchableOpacity>
      );
    };

    const { getByText } = render(
      <StripeTerminalProvider tokenProvider={tokenProvider} logLevel="error">
        <ChildImpl />
      </StripeTerminalProvider>
    );

    fireEvent.press(getByText('init'));
    expect(tokenProvider).toBeCalled();
    expect(tokenProvider).toReturnWith('_token');
  });
});
