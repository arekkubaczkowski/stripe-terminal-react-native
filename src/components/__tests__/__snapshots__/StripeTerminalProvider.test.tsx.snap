// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StripeTerminalProvider.tsx renders children correctly 1`] = `
<Text>
  test text
</Text>
`;
